test/test_KubeLibrary.py

```diff
@@ -577,6 +582,14 @@ def test_list_namespaced_secret_by_pattern(self, mock_lnp):
         self.assertEqual(kl.filter_names(secrets), kl.filter_names(secrets2))
         self.assertEqual(['grafana'], kl.filter_names(secrets))
 
+    @mock.patch('kubernetes.stream.stream')
+    def test_get_namespaced_exec(self, mock_stream):
+        test_string = "This is test String!"
+        mock_stream.return_value = test_string
+        kl = KubeLibrary(kube_config='test/resources/k3d')
+        stdout = kl.get_namespaced_pod_exec(name="pod_name", namespace="default", cmd=f"echo {test_string}")
+        self.assertEqual(stdout, test_string)
```

Comment:
I would change this test to add some stream handling, like iterating over it (with more than one line), from what I saw the stream is iterable, you could test how this can be handled, and possibly check how robotframework behaves (didn't check). It would be nice to show in examples how to approach it.

Reply:
I do not see any other option for now as get_namespaced_pod_exec always returns the string value.
Let me know if you have an idea on that :)

Reply:
Ok I need to take closer look, from what I understood stream works as generator, so for longer running commands the result might be splitted (not produced in one shot) but need to check.